ui/chapter/Chapter.tsx

```diff
@@ -47,7 +48,13 @@ export default function Chapter({ children, metadata, lang }) {
   const chapter = chapters[metadata.slug]
   const position = metadata.position + 1
   const isEven = position % 2 == 0
-
+  const chapterLessons = keys.filter((ele) =>
+    ele.includes(progress.substring(0, 3))
+  )
+  const isBetweenChapter =
+    progress !== chapterLessons[0] &&
+    progress !== chapterLessons.pop() &&
```

Comment:
With the chapterLessons.pop() being equal to outro-2 if i quit the game on the end screen without continuing you progress remains at `CH1OUT2` but now i am prompted to restart the chapter

![image](https://github.com/saving-satoshi/saving-satoshi/assets/108441023/6425a7b9-2475-4069-92b7-9202255ad4ce)

Reply:
I think you should remove the logic where `progress !== chapterLessons.pop()` until we add some logic that allows the outro screen to be automatically saved / skipped

Reply:
If we want the logic to allow me to quit at the outro screen we need to add some logic that allows quitting at the outro screen to also save or skip progress to the next lesson. However this should only be acheiveable if the user is logged in as they could jump past the logging in logic created for future lessons. Until then, I do not think that `progress !== chapterLessons.pop()` will work as expected

Reply:
With my progress at `CH1OUT2` I will still need to progress forward before the next chapter is unlocked, we can add some logic so that the outro screen progress is automatically skipped but we also need to consider in the first chapter we are expecting/requiring the user to log in so this logic would need to check if the user is logged in first

Reply:
> With the chapterLessons.pop() being outro-2 and my progress at CH1OUT2 if i quit the game on the end screen without continuing you progress remains at CH1OUT2 but now i am prompted to restart the chapter

You are right! That's an unintended behavior. Well, now I know what happens when you split development between two separate days 😅

> Until then, I do not think that progress !== chapterLessons.pop() will work as expected

I agree with your reasoning. Let me go ahead and remove this part of the code with the next update of PR.